runtime/src/staking.rs

```diff
@@ -0,0 +1,86 @@
+use crate::Perbill;
+
+use codec::{Decode, EncodeLike};
+use frame_support::{traits::Currency, dispatch::{DispatchResultWithPostInfo, Vec}};
+use frame_system::{Config as SysConfig, RawOrigin};
+use pallet_staking::Pallet as StakingPallet;
+use pallet_utxo::staking::StakingHelper;
+use sp_core::{H256, sp_std::vec, sr25519::Public};
+use sp_runtime::traits::StaticLookup;
+use sp_runtime::AccountId32;
+
+type StakeAccountId<T> =  <T as SysConfig>::AccountId;
+type LookupSourceOf<T> = <<T as SysConfig>::Lookup as StaticLookup>::Source;
+
+pub struct StakeOps<T>(sp_core::sp_std::marker::PhantomData<T>);
+impl <T: pallet_staking::Config + pallet_utxo::Config + pallet_session::Config> StakingHelper<T::AccountId> for StakeOps<T>
+    where StakeAccountId<T>: From<Public> + EncodeLike<AccountId32>
+{
+    fn get_account_id(pub_key: &H256) -> StakeAccountId<T> {
+      Public::from_h256(pub_key.clone()).into()
```

Comment:
Yes. :( This is because in the `type AccountId` of `runtime/lib.rs`, 
This portion here
```
/// Some way of identifying an account on the chain. We intentionally make it equivalent
/// to the public key of our transaction signing scheme.
pub type AccountId = <<Signature as Verify>::Signer as IdentifyAccount>::AccountId;
```

The `AccountId` is actually **`AccountId32`**.  It points to `sp_runtime::primitives`:
```
impl traits::IdentifyAccount for MultiSigner {
	type AccountId = AccountId32;
	fn into_account(self) -> AccountId32 {
		match self {
			Self::Ed25519(who) => <[u8; 32]>::from(who).into(),
			Self::Sr25519(who) => <[u8; 32]>::from(who).into(),
			Self::Ecdsa(who) => sp_io::hashing::blake2_256(who.as_ref()).into(),
		}
	}
}
```

Reply:
Can this not work with pk directly? AccountID is just pk anyway right? How much work would be involved?

Reply:
ah, yes, I'll change it from `From<Public>` to **`From<[u8; 32]>`**, so the function will just be:
```
    fn get_account_id(pub_key: &H256) -> StakeAccountId<T> {
        pub_key.0.into()
    }
```

Reply:
That sounds like a good plan